Switch to SVG icon

NotMyFault opened the pull request:
A simple change that integrates well into Jenkins modern UI and tables based on my quick testing.
cc @jglick If you don't mind taking a look :)

<details>
<summary>Job creation light theme</summary>

![](https://i.imgur.com/e1QD5F7.png)

</details>

<details>
<summary>Job creation dark theme</summary>

![](https://i.imgur.com/HqUHEJB.png)

</details>

<details>
<summary>Comparison with current look</summary>

![](https://i.imgur.com/mNF2XHa.png)

</details>

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

jglick reviewed (commented):
Why the addition of `doc/images/multi-configuration-project.png`?

NotMyFault commented:
> Why the addition of `doc/images/multi-configuration-project.png`?

![](https://i.imgur.com/XsErFqY.png)

I've exchanged the existing screenshot with the new icon. Looks like GitHub handles diffs for images differently now, I removed it and replaced it.
